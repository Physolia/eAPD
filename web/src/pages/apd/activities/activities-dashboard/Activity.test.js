import { shallow } from 'enzyme';
import React from 'react';

import { removeActivity } from '../../../../actions/editActivity';

import {
  plain as EntryDetails,
  mapStateToProps,
  mapDispatchToProps
} from './Activity';

const mockPush = jest.fn();

jest.mock('react-router-dom', () => {
  return {
    useHistory: jest.fn().mockReturnValue({ push: () => mockPush() }),
    useRouteMatch: jest.fn().mockReturnValue({ path: '---path---' })
  };
});

describe('the (Activity) EntryDetails component', () => {
  const props = {
    apdId: '0123456789abcdef01234567',
    activityIndex: 2,
    activityKey: 'activity key',
    fundingSource: 'money pit',
    jumpAction: jest.fn(),
    name: 'activity name',
    remove: jest.fn()
  };

  beforeEach(() => {
    if (mockPush) {
      mockPush.mockReset();
    }
    props.jumpAction.mockReset();
    props.remove.mockClear();
  });

  test('renders correctly with no activity name', () => {
    expect(shallow(<EntryDetails {...props} name={null} />)).toMatchSnapshot();
  });

  test('renders correctly with no activity funding source/program', () => {
    expect(
      shallow(<EntryDetails {...props} fundingSource={null} />)
    ).toMatchSnapshot();
  });

  test('renders correctly with no activity name or funding source/program', () => {
    expect(
      shallow(<EntryDetails {...props} fundingSource={null} name={null} />)
    ).toMatchSnapshot();
  });

  test('does not render the delete button on the first element', () => {
    const firstActivityProps = {
      apdId: '0123456789abcdef01234567',
      activityIndex: 0,
      activityKey: 'activity 1 key',
      fundingSource: 'money pit',
      jumpAction: jest.fn(),
      name: 'activity 1 name',
      remove: jest.fn()
    };
    const component = shallow(<EntryDetails {...firstActivityProps} />);
    expect(component).toMatchSnapshot();
  });

  test('deletes the element', () => {
    const component = shallow(<EntryDetails {...props} />);
    const review = component.find('Review').dive();
    // Second button is the delete button
    review.find('Button').at(1).simulate('click');
    // The remove function should not have fired.
    expect(props.remove).toHaveBeenCalledTimes(0);
  });

  test('maps state to props', () => {
    const state = {
      apd: {
        data: {
          activities: [
            {
              fundingSource: 'money pit',
              key: 'key1',
              name: 'that free money guy'
            },
            {
              fundingSource: 'black market kidneys',
              key: 'key2',
              name: 'scary alley'
            },
            {
              fundingSource: 'appropriations',
              key: 'key3',
              name: 'Congress Dollars'
            },
            {
              fundingSource: 'blackjack',
              key: 'key4',
              name: 'Lucky Pete'
            }
          ]
        }
      }
    };

    expect(mapStateToProps(state, { activityIndex: 2 })).toEqual({
      activityKey: 'key3',
      fundingSource: 'appropriations',
      name: 'Congress Dollars'
    });
  });

  test('maps dispatch to props', () => {
    expect(mapDispatchToProps).toEqual({
      remove: removeActivity
    });
  });
});
