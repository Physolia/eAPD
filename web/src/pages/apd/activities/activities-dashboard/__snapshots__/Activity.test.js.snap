// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the (Activity) EntryDetails component does not render the delete button on the first element 1`] = `
<div
  className="activity--body activity--body__first"
  id="activity-activity 1 key"
>
  <Review
    className="entry-details--review"
    editContent={
      <div
        className="nowrap visibility--screen"
      >
        <Button
          aria-label="Edit: Activity 1: Activity 1 Name (Money Pit)"
          component={[Function]}
          href="/apd/0123456789abcdef01234567/activity/0/overview"
          size="small"
          type="button"
          variation="transparent"
        >
          Edit
        </Button>
      </div>
    }
    editText="Edit"
    heading="Activity 1: Activity 1 Name (Money Pit)"
    headingLevel="4"
  />
</div>
`;

exports[`the (Activity) EntryDetails component renders correctly with no activity funding source/program 1`] = `
<div
  className="activity--body activity--body__notfirst"
  id="activity-activity key"
>
  <Review
    className="entry-details--review"
    editContent={
      <div
        className="nowrap visibility--screen"
      >
        <Button
          aria-label="Edit: Activity 3: Activity Name"
          component={[Function]}
          href="/apd/0123456789abcdef01234567/activity/2/overview"
          size="small"
          type="button"
          variation="transparent"
        >
          Edit
        </Button>
        <React.Fragment>
          <span>
            |
          </span>
          <Button
            aria-label="Delete: Activity 3: Activity Name"
            className="ds-u-color--error"
            component="button"
            onClick={[Function]}
            size="small"
            type="button"
            variation="transparent"
          >
            Delete
          </Button>
        </React.Fragment>
      </div>
    }
    editText="Edit"
    heading="Activity 3: Activity Name"
    headingLevel="4"
  />
</div>
`;

exports[`the (Activity) EntryDetails component renders correctly with no activity name 1`] = `
<div
  className="activity--body activity--body__notfirst"
  id="activity-activity key"
>
  <Review
    className="entry-details--review"
    editContent={
      <div
        className="nowrap visibility--screen"
      >
        <Button
          aria-label="Edit: Activity 3: Untitled (Money Pit)"
          component={[Function]}
          href="/apd/0123456789abcdef01234567/activity/2/overview"
          size="small"
          type="button"
          variation="transparent"
        >
          Edit
        </Button>
        <React.Fragment>
          <span>
            |
          </span>
          <Button
            aria-label="Delete: Activity 3: Untitled (Money Pit)"
            className="ds-u-color--error"
            component="button"
            onClick={[Function]}
            size="small"
            type="button"
            variation="transparent"
          >
            Delete
          </Button>
        </React.Fragment>
      </div>
    }
    editText="Edit"
    heading="Activity 3: Untitled (Money Pit)"
    headingLevel="4"
  />
</div>
`;

exports[`the (Activity) EntryDetails component renders correctly with no activity name or funding source/program 1`] = `
<div
  className="activity--body activity--body__notfirst"
  id="activity-activity key"
>
  <Review
    className="entry-details--review"
    editContent={
      <div
        className="nowrap visibility--screen"
      >
        <Button
          aria-label="Edit: Activity 3: Untitled"
          component={[Function]}
          href="/apd/0123456789abcdef01234567/activity/2/overview"
          size="small"
          type="button"
          variation="transparent"
        >
          Edit
        </Button>
        <React.Fragment>
          <span>
            |
          </span>
          <Button
            aria-label="Delete: Activity 3: Untitled"
            className="ds-u-color--error"
            component="button"
            onClick={[Function]}
            size="small"
            type="button"
            variation="transparent"
          >
            Delete
          </Button>
        </React.Fragment>
      </div>
    }
    editText="Edit"
    heading="Activity 3: Untitled"
    headingLevel="4"
  />
</div>
`;
